src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+type Props = {
+  filteredTodos: Todo[];
+  tempTodo: Todo | null;
+  onDelete: (id: number) => void;
+  deletingTodoId: number | null;
+};
+
+export const TodoList: React.FC<Props> = ({
+  filteredTodos,
+  tempTodo,
+  onDelete,
+  deletingTodoId,
+}) => {
+  return (
+    <>
+      {filteredTodos.map(todo => (
+        <TodoItem
+          todo={todo}
+          key={todo.id}
+          onDelete={onDelete}
+          isDeleting={deletingTodoId === todo.id}
+        />
+      ))}
+
+      <TodoItem
+        todo={tempTodo}
+        onDelete={onDelete}
+        isDeleting={deletingTodoId === tempTodo?.id}
```

Comment:
You can create a separate component for tempTodo 
<img width="628" alt="image" src="https://github.com/user-attachments/assets/2f2ed1da-faa6-4fb1-9035-8d7803bb4bf8" />